### ISSUE TITLE
TSQL CREATE TYPE <x> FROM <BASETYPE>(<precision>[, <scale>]) fails to parse

### ISSUE THREAD

**Issue by credfeto** [sqlfluff/sqlfluff#7583]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

Parsing error when using ``CREATE TYPE [dbo].[TestType] FROM VARCHAR(50);``

similar if type is ``DECIMAL(38, 10)``

### Expected Behaviour

File to parse and understand the precision and scale of aliases of base types.

### Observed Behaviour

Gives the following errors:
```
== [Test.sql] FAIL                                                                                                                                                                                                                                                       
L:   4 | P:  45 |  PRS | Line 4, Position 45: Found unparsable section:
                       | '(50);\nGO'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

if its just an alias to VARCHAR with no precision then its happy.

### How to reproduce

```bash
sqlfluff lint test.sql
```

test.sql
```tsql
SET QUOTED_IDENTIFIER ON;
SET ANSI_NULLS ON;

CREATE TYPE [dbo].[ExampleType] FROM VARCHAR(50);
GO
```


### Dialect

tsql

### Version

4.0.4

Python 3.14.3

### Configuration

[sqlfluff]
dialect = tsql



### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
